### ISSUE TITLE
Allow excluding subdirectories like node_modules

### ISSUE THREAD

**Issue by ssorallen** [facebook/jscodeshift#53]
I'd like to run a mod on a large directory of directories, each of which can have a 'node_modules' directory I'd like to ignore. It'd be great to have an option in jscodeshift to exclude subdirectory patterns if you pass it a directory to transform.

**Comment by fkling** [comment]
In the meantime you could use `find` like so:

```
find ./ -name  *.js -not -path */node_modules/* | xargs jscodeshift
```

**Comment by ssorallen** [comment]
:+1: Works for me.

**Comment by DrewML** [comment]
I think this issue can be closed per #107.

**Comment by rattrayalex** [comment]
Summarizing the fix, to do all files in a repo from the root:

```
jscodeshift --ignore-config=".gitignore" --ignore-pattern="**/node_modules/**" -t path/to/my_transform.js .
```

Note that the gitignore support is a bit shoddy because it doesn't use the same glob syntax, so for example `node_modules` won't match subdirectories (hence its additional inclusion here, with double asterixes).
